File: Data Structures & Algorithms/combinations/submission-0.js
class Solution {
    /**
     * @param {number} n
     * @param {number} k
     * @return {number[][]}
     */
    combine(n, k) {

        const res = [];
        const nums= [];

        function calculate(totalNums,index){
            if(totalNums === k) {
                res.push([...nums]);
                return;
            }

            for(let i = index; i<= n ;i++){
                nums.push(i);
                calculate(totalNums+1, i+1);
                nums.pop();
            }
        }

        calculate(0, 1);

        return res;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def dfs(pairs:List[int],index:int):
            if index == len(nums):
                res.append(pairs)
                return
            dfs(pairs,index+1)
            dfs(pairs + [nums[index]],index+1)
        
        dfs([],0)
        return res
        
        